packages/extension-ui/src/Popup/Signing/Signing.test.tsx

```diff
@@ -0,0 +1,194 @@
+        specVersion: '0x00000070',
+        tip: '0x00000000000000000000000000000000',
+        version: 1
+      }
+    }];
+    onActionStub = jest.fn();
+    wrapper = mountComponent();
+    await act(flushAllPromises);
+    wrapper.update();
+  });
+
+  describe('Switching between requests', () => {
+    it('initially first request should be shown', () => {
+      expect(wrapper.find(TransactionIndex).text()).toBe('Transaction:1/2');
+      expect(wrapper.find(Request).prop('signId')).toBe(signRequests[0].id);
+    });
+
+    it('only ArrowRight should be active on first screen', () => {
+      expect(wrapper.find('ArrowLeft').prop('isActive')).toBe(false);
+      expect(wrapper.find('ArrowRight').prop('isActive')).toBe(true);
+      wrapper.find('ArrowLeft').simulate('click');
+      expect(wrapper.find(TransactionIndex).text()).toBe('Transaction:1/2');
+    });
+
+    it('should display second request after clicking right arrow', () => {
+      wrapper.find('ArrowRight').simulate('click');
+      expect(wrapper.find(TransactionIndex).text()).toBe('Transaction:2/2');
+      expect(wrapper.find(Request).prop('signId')).toBe(signRequests[1].id);
+    });
+
+    it('only ArrowLeft should be active on second screen', () => {
+      wrapper.find('ArrowRight').simulate('click');
+      expect(wrapper.find('ArrowLeft').prop('isActive')).toBe(true);
+      expect(wrapper.find('ArrowRight').prop('isActive')).toBe(false);
+      expect(wrapper.find(TransactionIndex).text()).toBe('Transaction:2/2');
+    });
+
+    it('should display previous request after ArrowLeft has been clicked', () => {
+      wrapper.find('ArrowRight').simulate('click');
+      wrapper.find('ArrowLeft').simulate('click');
+      expect(wrapper.find(TransactionIndex).text()).toBe('Transaction:1/2');
+      expect(wrapper.find(Request).prop('signId')).toBe(signRequests[0].id);
+    });
+  });
+
+  describe('Request rendering', () => {
+    it('correctly displays request 1', () => {
+      expect(wrapper.find(Address).text()).toBe('acc15D4bqjQRPgdMBK8bNvhX4tSuCtSGZS7rZjD5XH5SoKcFeKn5');
+      expect(wrapper.find(Details).find('td.data').map((el): string => el.text())).toEqual([
+        'polkadot.js',
+        'Alexander',
+        '112',
+        '0',
+        'balances.transfer',
+        `{
+  "dest": "5D4bqjQRPgdMBK8bNvhX4tSuCtSGZS7rZjD5XH5SoKcFeKn5",
+  "value": 31415926500000
+}`,
+        ' Transfer some liquid free balance to another account.   `transfer` will set the `FreeBalance` of the sender and receiver.  It will decrease the total issuance of the system by the `TransferFee`.  If the sender\'s account is below the existential deposit as a result  of the transfer, the account will be reaped.   The dispatch origin for this call must be `Signed` by the transactor.',
+        'mortal, valid from #4,033,610 to #4,033,674'
+      ]);
+    });
+
+    it('correctly displays request 2', () => {
+      wrapper.find('ArrowRight').simulate('click');
```

Comment:
Not sure if this test adds any value. If you want to test-drive that the UI values are not hard-coded, you can either pass in 1 different request. Or navigate to the second one, but test just 1-2 values.